src/charm.py

```diff
@@ -143,6 +144,38 @@ def snap_path(self) -> Optional[str]:
 
         return self._snap_path
 
+    @property
+    def snap_channel(self) -> str:
+        """Get the channel for exporter snap.
+
+        The channel is determined by the controller version that the charm is deployed
+        under. In case the controller version is less than 2.9, the snap channel is set
+        to 2.8/stable. Otherwise, the snap channel is set to 2.9/stable.
+        """
+        controller_version = self.get_controller_version()
+
+        if controller_version < version.parse("2.9"):
+            return "2.8/stable"
+        if (
+            controller_version.major == version.parse("2.9").major
+            and controller_version.minor == version.parse("2.9").minor
+        ):
+            return "2.9/stable"
+
+        return "latest/stable"
```

Comment:
The `latest/stable` right now is version 1.0.4 and that is suitable for controller 2.8, or not? I think it would be better to raise an exception that controller 3.0 above are not supported yet.

Reply:
`latest/stable`'s current version is 2.0.0, which is the same as `2.9/stable` (you are seeing 1.0.4 probably because of the cache). But you are right, the safer way would be to raise exceptions for unsupported controller versions. Changed.